.dat file viewer

nptr opened the pull request:
In order to get used to the project etc., i started playing around with the tooling a bit.
I kind of cloned the DatFileTester and improved the UI and display of the images.

Maybe its useful :)

andreas-eberle commented:
@nptr: Thanks for this! It's much easier to see the graphics with this. 

I have two requests: 
* Can you tune the code a bit to get rid of all the warnings (I use IntelliJ IDEA)?
* Can you reset the viewpoint when a new .dat file is selected? Beause I just scrolled a bit around and selected the next file and had to manually get back to the start position.

nptr commented:
@andreas-eberle 

* removed most of the warnings IntelliJ gave me. I left two "lambda with unnecessary code block" warnings to have the code consistent, and also the disclaimer formatting since thats how its used in the rest of the project.
* viewport + zoomlevel reset on file change
* fixed file selection issue that made it look like the canvas isn't cleared (was actually loading the same file again sometimes)

i hope its good now - for a dev tool :)

nptr commented on jsettlers.tools/src/main/java/jsettlers/graphics/debug/DatViewer/DatFileViewer.java:
```diff
@@ -0,0 +1,500 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *******************************************************************************/
+package jsettlers.graphics.debug.DatViewer;
```
true, fixed it.
